cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/ParameterJsonStructureTest.java

```diff
@@ -0,0 +1,141 @@
+ */
+package com.linkedin.kafka.cruisecontrol.servlet.parameters;
+
+import com.linkedin.cruisecontrol.servlet.parameters.CruiseControlParameters;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.Paths;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.parameters.Parameter;
+import io.swagger.v3.parser.OpenAPIV3Parser;
+
+import io.swagger.v3.parser.core.models.ParseOptions;
+import io.swagger.v3.parser.core.models.SwaggerParseResult;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Set;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.TreeSet;
+
+public class ParameterJsonStructureTest {
+  private Map<String, Class> _endpointToClass;
+  OpenAPI _openAPI;
+  /**
+   * Specify endpoints to be tested
+   */
+  @Before
+  public void setupParameterClasses() {
+    _endpointToClass = new HashMap<>();
+    _endpointToClass.put("/kafkacruisecontrol/rebalance", RebalanceParameters.class);
+    _endpointToClass.put("/kafkacruisecontrol/partition_load", PartitionLoadParameters.class);
+  }
+
+  /**
+   * Load the OpenAPI files for endpoints and compares them against the source code
+   */
+  @Test
+  public void loadOpenApiSpec() throws Exception {
+    OpenAPIV3Parser openApiParser = new OpenAPIV3Parser();
+    String baseFileName = "../schemas/base.yaml";
+    ParseOptions options = new ParseOptions();
+    options.setResolve(true);
+    options.setFlatten(true);
+
+    SwaggerParseResult parseResult = openApiParser.readLocation(baseFileName, null, options);
+    _openAPI = parseResult.getOpenAPI();
+    Schema schema = parseSchema();
+    for (Map.Entry<String, Set<String>> endpoint: schema.getEndpointsAndParameters()) {
+      Assert.assertTrue(_endpointToClass.containsKey(endpoint.getKey()));
+      CruiseControlParameters endpointParams = (CruiseControlParameters) (_endpointToClass.get(endpoint.getKey()).newInstance());
+      Assert.assertEquals(
+        endpoint.getValue(),
+        endpointParams.caseInsensitiveParameterNames());
+    }
+  }
+
+  /**
+   * Return the list of parameters given the path item (an endpoint)
+   *
+   * @param pathItem Endpoint defined as a PathItem object
+   * @return set of parameters for the specified endpoint
+   */
+  public static Set<String> parseEndpoint(PathItem pathItem) throws IllegalArgumentException {
+    List<Parameter> parameterList;
+    Set<String> parameterSet = new TreeSet<>();
+    if (pathItem.getGet() != null) {
+      parameterList = pathItem.getGet().getParameters();
+    } else if (pathItem.getPost() != null) {
+      parameterList = pathItem.getPost().getParameters();
+    } else {
+      throw new IllegalArgumentException("Schema Parser does not support HTTP methods other than GET/POST");
+    }
+
+    for (Parameter parameter : parameterList) {
```

Comment:
I feel we should not deduplicate here. If in spec, the same parameter are defined twice or (name conflict between two parameters), it is a bug and the test is expected to fail.